infra/build_deploy_config.py

```diff
@@ -11,10 +11,31 @@
 infra_root = os.path.abspath(os.path.dirname(__file__))
 
 
+def validate_aws_config():
+    current_caller = boto3.client('sts').get_caller_identity()
+    if '@' not in current_caller['UserId'].split(':')[1]:
+        print('~/.aw/config needs to have an email under the role_session_name')
```

Comment:
Should this just raise?  For example (also including the suggestion from below to declare the variables in one place):
```
caller_info = boto3.client("sts").get_caller_identity()
owner = caller_info['UserId'].split(':')[1]

def validate_aws_config():
    if '@' not in owner:
        raise ValueError('~/.aw/config needs to have an email under the role_session_name')
```
The call would also be less code overall and just `validate_aws_config()`.

Reply:
i scrapped the function, and just threw it inline, with the raise, and var reuse.